lib/events/event_handler.ex

```diff
@@ -56,6 +61,46 @@ defmodule ElixirTools.Events.EventHandler do
     :ok
   end
 
+  @spec publish(Event.t(), event_schema, [events_opt]) :: :ok
+  def publish(event, schema, opts) do
+    task_supervisor_module = opts[:task_supervisor_module] || Task.Supervisor
+    not_sent_event_module = opts[:not_sent_event_module] || NotSentEvent
+
+    if valid_event?(schema, event) do
+      task_supervisor_module.async_nolink(ElixirTools.TaskSupervisor, fn ->
+        publish_event_call(event, opts)
+      end)
+    else
+      error_info =
+        event |> Map.from_struct() |> Map.put(:reason, "Event schema validation failed.")
+
+      not_sent_event_module.create!(%{content: Jason.encode!(error_info)})
+    end
+
+    :ok
+  end
+
+  @spec valid_event?(event_schema, Event.t()) :: boolean
```

Comment:
If you agree overall with this PR, I'll move this function to either a dedicated module or to `ElixirTools.Events.Event` (I prefer the second option).